tokio/src/sync/tests/loom_set_once.rs

```diff
@@ -0,0 +1,67 @@
+use crate::sync::SetOnce;
+
+use loom::future::block_on;
+use loom::sync::atomic::AtomicU32;
+use loom::thread;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+
+struct Foo {
+    pub num_drops: Arc<AtomicU32>,
+}
+
+impl Foo {
+    pub fn new(num_drops: Arc<AtomicU32>) -> Self {
+        Self { num_drops }
+    }
+}
+
+impl Drop for Foo {
+    fn drop(&mut self) {
+        self.num_drops.fetch_add(1, Ordering::Release);
+    }
+}
+
+#[test]
+fn set_once_drop_test() {
+    loom::model(|| {
+        let tx = Arc::new(SetOnce::new());
+        let rx = Arc::clone(&tx);
+
+        let drop_counter = Arc::new(AtomicU32::new(0));
+        let counter_cl = Arc::clone(&drop_counter);
+
+        let thread = thread::spawn(move || {
+            return rx.set(Foo::new(counter_cl)).is_ok();
+        });
+
+        let foo = Foo::new(Arc::clone(&drop_counter));
+
+        let set = tx.set(foo).is_ok();
+        let res = thread.join().unwrap();
+
+        drop(tx);
+
+        assert_eq!(drop_counter.load(Ordering::Acquire), 2);
```

Comment:
For reviewers: we don't need RMW here because the

1. `Arc` invokes `Foo::drop` in the current thread.
2. AND the `Foo::drop` performs an `Release` RMW.